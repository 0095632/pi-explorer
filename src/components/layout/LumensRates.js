import React from 'react'
import 'whatwg-fetch'

const FEED_URL = 'https://api.coinmarketcap.com/v1/ticker/stellar/'

class LumensRatesContainer extends React.Component {
  componentDidMount() {
    fetch(FEED_URL).then((rsp) => rsp.json()).then((rspJson) => {
      const lumens = rspJson[0]
      const newState = {
        btc: lumens.price_btc,
        change: lumens.percent_change_24h,
        usd: lumens.price_usd
      }
      this.setState(newState)
    }).catch((err) => {
      console.error(`Failed to fetch price: [${err}]`)
    })
  }
  render() {
    if (!this.state)
      return null
    return <LumensRates {...this.state}/>
  }
}

class LumensRates extends React.Component {
  isPositive(changeNumStr) {
    const asFloat = Number.parseFloat(changeNumStr)
    return (Number.isNaN(asFloat) === false && Number(asFloat) >= 0)
  }

  renderChange(change) {
    const positive = this.isPositive(change)
    const valueStr = `${ (positive)
      ? '+'
      : '-'}${this.props.change}%`
    const style = {
      color: (positive)
        ? 'green'
        : 'red'
    }
    return <span>Change (24hrs):&nbsp;
      <span style={style}>{valueStr}</span>
    </span>
  }

  render() {
    return (
      <div>
        <span>USD: {this.props.usd}</span>
        &nbsp;|&nbsp;
        <span>BTC: {this.props.btc}</span>
        &nbsp;|&nbsp; {this.renderChange(this.props.change)}
      </div>
    )
  }
}

export default LumensRatesContainer
