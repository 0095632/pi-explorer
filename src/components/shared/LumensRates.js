import React from 'react'
import 'whatwg-fetch'

const FEED_URL = 'https://api.coinmarketcap.com/v1/ticker/stellar/'

class LumensRatesContainer extends React.Component {
  componentDidMount() {
    fetch(FEED_URL)
      .then(rsp => rsp.json())
      .then(rspJson => {
        const lumens = rspJson[0]
        const newState = {
          btc: lumens.price_btc,
          change: lumens.percent_change_24h,
          usd: lumens.price_usd,
        }
        this.setState(newState)
      })
      .catch(err => {
        console.error(`Failed to fetch price: [${err}]`)
      })
  }
  render() {
    if (!this.state) return null
    return <LumensRates {...this.state} />
  }
}

class LumensRates extends React.Component {
  isPositive(changeNumStr) {
    const asFloat = Number.parseFloat(changeNumStr)
    return Number.isNaN(asFloat) === false && Number(asFloat) >= 0
  }

  renderChange(change) {
    const positive = this.isPositive(change)
    const valueStr = `${positive ? '+' : ''}${this.props.change}%`
    const style = {
      color: positive ? 'green' : 'red',
    }
    return (
      <span>
        <span style={style}>
          {valueStr}
        </span>
      </span>
    )
  }

  render() {
    return (
      <span>
        XLM/USD: {this.props.usd} {this.renderChange(this.props.change)}
      </span>
    )
  }
}

export default LumensRatesContainer
