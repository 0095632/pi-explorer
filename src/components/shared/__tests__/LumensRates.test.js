import React from 'react'
import {shallow} from 'enzyme'
import {LumensRates} from '../LumensRates'

it('render positive change in rate', () => {
  const rate = shallow(<LumensRates change="2.1" usd="0.020" />)
  const changeEl = rate.find('span[style]')
  expect(changeEl.props().style.color).toEqual('green')
  expect(rate.getNodes()).toMatchSnapshot()
})

it('render negative change in rate', () => {
  const rate = shallow(<LumensRates change="-1.52" usd="0.025" />)
  const changeEl = rate.find('span[style]')
  expect(changeEl.props().style.color).toEqual('red')
  expect(rate.getNodes()).toMatchSnapshot()
})
