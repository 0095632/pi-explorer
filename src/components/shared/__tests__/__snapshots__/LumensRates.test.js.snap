// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render negative change in rate 1`] = `
Array [
  <span>
    XLM/USD: 
    0.025
     
    <span
        style={
            Object {
                "color": "red",
              }
        }
    >
        -1.52%
    </span>
</span>,
]
`;

exports[`render positive change in rate 1`] = `
Array [
  <span>
    XLM/USD: 
    0.020
     
    <span
        style={
            Object {
                "color": "green",
              }
        }
    >
        +2.1%
    </span>
</span>,
]
`;
